services/ui-src/src/components/fields/ChoiceListField.tsx

```diff
@@ -83,6 +80,28 @@ export const ChoiceListField = ({
     });
   };
 
+  const clearNestedValues = (choices: FieldChoice[]) => {
+    return choices.forEach((choice: FieldChoice) => {
+      // if a choice is not selected and there are children, clear out any saved data
+      if (!choice.checked && choice.children) {
+        choice.children.map((child) => {
+          if (child.type === "radio" || child.type === "checkbox") {
+            form.setValue(child.id, [], { shouldValidate: true });
+            if (child.props?.choices) {
+              child.props.choices.forEach((choice: FieldChoice) => {
+                choice.checked = false;
+              });
+              clearNestedValues(child.props.choices);
+            }
+          } else {
+            form.setValue(child.id, "", { shouldValidate: true });
+          }
+        });
```

Comment:
This might be a good use case for a switch statement so it reads a bit easier. We also want to support more nested field types in the future (dropdown, single checkbox -- Jira ticket here: https://qmacbis.atlassian.net/browse/MDCT-2005) so this might be a step in that direction.